src/punycode.cpp

```diff
@@ -10,6 +10,16 @@ namespace Url
 
     std::string& Punycode::encode(std::string& str)
     {
+        std::string output;
+        encode(output, str.cbegin(), str.cend());
+        str.assign(output);
```

Comment:
For C++, I'd just use the assignment operator. I don't know if this has been rebased to include that change or not.